### PR TITLE
PyTorch: From Centralized To Federated

### PR DESCRIPTION
**Pull request by mariaboerner1987**
Create README and flower.dev doc for PyTorch Centralized and Federated setup.

### CODE REVIEW

**Comment by mariaboerner1987** [comment]
correction: the last commit change the pytorch_minimal example from source to example directory

File: doc/source/example-pytorch-from-centralized-to-federated.rst [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+
+    python3 cifar.py
+
+
+Federated Training
+------------------
+
+The concept of centralized learning as shown in the previous section is known for most of you and many of you have set it up already. Normally, if you want to run machine learning workloads federated you have to change your complete code and set up everything from scratch. This is quite a big effort. 
+
+However, if you use Flower you can keep your code and put Flower on top. You can play around with federated learning setups without destroying your actual pre-existing code.
```
Comment (tanertopal) [inline doc/source/example-pytorch-from-centralized-to-federated.rst:171]: Just an idea. How about:

```suggestion
However, with Flower you can evolve your pre-existing code into a federated learning setup without the need for a major rewrite.
```

File: doc/source/example-pytorch-from-centralized-to-federated.rst [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    import numpy as np
+    import torch
+
+    import cifar
+    import flwr as fl
+
+    DEVICE: str = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+Implementing a Flower *client* basically means implementing a subclass of either :code:`flwr.client.Client` or :code:`flwr.client.NumPyClient`.
+Our implementation will be based on :code:`flwr.client.NumPyClient` and we'll call it :code:`CifarClient`.
```
Comment (tanertopal) [inline doc/source/example-pytorch-from-centralized-to-federated.rst:212]:
```suggestion
Our implementation will be based on :code:`flwr.client.NumPyClient` and we'll call it :code:`CifarClient`. As a side note the NumPyClient helps avoiding some boilerplate code which is required in many cases.
```

File: examples/pytorch_from_centralized_to_federated/server.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+"""Flower server example."""
+
+
+import flwr as fl
+
+if __name__ == "__main__":
+    fl.server.start_server("[::]:8080", config={"num_rounds": 3})
```
Comment (tanertopal) [inline examples/pytorch_from_centralized_to_federated/server.py:7]:
```suggestion
    fl.server.start_server(config={"num_rounds": 3})
```

File: examples/pytorch_from_centralized_to_federated/client.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+def main() -> None:
+    """Load data, start CifarClient."""
+
+    # Load model and data
+    model = cifar.Net().to(DEVICE)
+    trainloader, testloader = cifar.load_data()
+
+    # Start client
+    client = CifarClient(model, trainloader, testloader)
+    fl.client.start_numpy_client("[::]:8080", client)
```
Comment (tanertopal) [inline examples/pytorch_from_centralized_to_federated/client.py:71]:
```suggestion
    fl.client.start_numpy_client("0.0.0.0:8080", client)
```

File: doc/source/example-pytorch-from-centralized-to-federated.rst [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+        """Load data, start CifarClient."""
+
+        # Load model and data
+        model = cifar.Net()
+        model.to(DEVICE)
+        trainloader, testloader = cifar.load_data()
+
+        # Start client
+        client = CifarClient(model, trainloader, testloader)
+        fl.client.start_numpy_client("[::]:8080", client)
```
Comment (tanertopal) [inline doc/source/example-pytorch-from-centralized-to-federated.rst:290]:
```suggestion
        fl.client.start_numpy_client("0.0.0.0:8080", client)
```

**Review by danieljanes — APPROVED** [review]

**Pull request merged**